vowpalwabbit/object_pool.h

```diff
@@ -112,7 +113,6 @@ struct no_lock_object_pool
 
     for (size_t i = 0; i < size; i++)
     {
-      memset(&chunk[i], 0, sizeof(T));
       new (&chunk[i]) T{};
```

Comment:
You don't need to call new on each chunk here, you're doing it a few lines earlier

Reply:
Right, I had added this before I realized unique_ptr<T[]> calls the constructor, thanks for the catch!